services/src/api/model/operators.rs

```diff
@@ -27,21 +32,91 @@ pub struct RasterResultDescriptor {
     pub data_type: RasterDataType,
     #[schema(value_type = String)]
     pub spatial_reference: SpatialReferenceOption,
-    pub measurement: Measurement,
     pub time: Option<TimeInterval>,
     pub bbox: Option<SpatialPartition2D>,
     pub resolution: Option<SpatialResolution>,
+    pub bands: RasterBandDescriptors,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, ToSchema)]
+pub struct RasterBandDescriptors(Vec<RasterBandDescriptor>);
+
+impl RasterBandDescriptors {
+    pub fn new(bands: Vec<RasterBandDescriptor>) -> Result<Self> {
+        let mut names = HashSet::new();
+        for value in &bands {
+            ensure!(!value.name.is_empty(), RasterBandNameMustNotBeEmpty);
+            ensure!(value.name.byte_size() <= 256, RasterBandNameTooLong);
+            ensure!(
+                names.insert(&value.name),
+                RasterBandNamesMustBeUnique {
+                    duplicate_key: value.name.clone()
+                }
+            );
+        }
+
+        Ok(Self(bands))
+    }
+}
+
+impl<'de> Deserialize<'de> for RasterBandDescriptors {
```

Comment:
can't we derive that?

Reply:
I check for duplicates etc. during deserialization.